canal/sync.go

```diff
@@ -77,10 +77,16 @@ func (c *Canal) runSyncBinlog() error {
 		case *replication.RowsEvent:
 			// we only focus row based event
 			err = c.handleRowsEvent(ev)
-			if err != nil && errors.Cause(err) != schema.ErrTableNotExist {
-				// We can ignore table not exist error
-				log.Errorf("handle rows event at (%s, %d) error %v", pos.Name, curPos, err)
-				return errors.Trace(err)
+			if err != nil {
+				realErr := errors.Cause(err)
+				// if error is ErrTableNotExist or ErrMissingTableMeta, skip this event
+				if realErr == schema.ErrTableNotExist || (c.cfg.DiscardNoMetaRowEvent && realErr == schema.ErrMissingTableMeta) {
```

Comment:
you can remove c.cfg.DiscardNoMetaRowEvent  check here.

Reply:
done